src/Input/InputParamTrait.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+/**
+ * @see       https://github.com/laminas/laminas-cli for the canonical source repository
+ * @copyright https://github.com/laminas/laminas-cli/blob/master/COPYRIGHT.md
+ * @license   https://github.com/laminas/laminas-cli/blob/master/LICENSE.md New BSD License
+ */
+
+declare(strict_types=1);
+
+namespace Laminas\Cli\Input;
+
+use ArrayObject;
+use InvalidArgumentException;
+use Symfony\Component\Console\Helper\QuestionHelper;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+use function sprintf;
+
+trait InputParamTrait
+{
+    /**
+     * @internal
+     * @var null|ArrayObject<string, InputParam>
+     */
+    private $inputParams;
+
+    /**
+     * @param null|mixed $default
+     * @return $this
+     */
+    final public function addParam(
+        string $name,
+        string $description,
+        string $type,
+        bool $required = false,
+        $default = null,
+        array $options = []
+    ) : self {
+        $mode = $type === InputParam::TYPE_BOOL
+            ? InputOption::VALUE_NONE
+            : InputOption::VALUE_REQUIRED;
+
+        $this->addOption(
+            $name,
+            null,
+            $mode,
+            $description
+            // default null, on purpose
+        );
+
+        if ($this->inputParams === null) {
```

Comment:
Considering the parent object composing the trait _could_ initialize the `$inputParams` property, it may be a good idea to raise an exception if it is non-null AND not an `ArrayObject`, considering you use the former's API here.